source/cl/test/UnitCL/include/cl_intel_unified_shared_memory.h

```diff
@@ -51,11 +60,109 @@ struct cl_intel_unified_shared_memory_Test : public virtual ucl::ContextTest {
 #undef GET_EXTENSION_ADDRESS
   }
 
+  void TearDown() override {
+    for (auto ptr : allPointers()) {
+      cl_int err = clMemBlockingFreeINTEL(context, ptr);
+      EXPECT_SUCCESS(err);
+    }
+    device_ptr = shared_ptr = host_ptr = nullptr;
+
+    ucl::ContextTest::TearDown();
+  }
+
   static void *getPointerOffset(void *ptr, size_t offset) {
     cl_char *byte_ptr = reinterpret_cast<cl_char *>(ptr) + offset;
     return reinterpret_cast<void *>(byte_ptr);
   }
 
+  /// Given a pointer, create a new USM allocation of the same type
+  ///
+  /// That is, if `old_ptr` points to a host USM allocation, this returns a new
+  /// host allocation. If the allocation type is unavailable (say, due to errors
+  /// querying the allocation data), nullptr is returned.
+  void *sameMemAlloc(void *old_ptr, size_t size, cl_uint alignment,
+                     cl_int *err) {
+    cl_int type_err;
+    cl_unified_shared_memory_type_intel alloc_type = CL_MEM_TYPE_UNKNOWN_INTEL;
+    type_err = clGetMemAllocInfoINTEL(context, old_ptr, CL_MEM_ALLOC_TYPE_INTEL,
```

Comment:
I'm not sure I see the problem with the extra testing. We could add infrastructure to store the types of all pointers, but that feels like it would involve a lot more code and complexity (what if a pointer is allocated in the body of a test instead of the initPointers method? What about offset pointers?).

All our tests also "test" things like allocating and freeing memory, so I think another function isn't that big a deal.

Reply:
I think the purer the test the better, so we can identify the issues better. Also, by adding API calls, you can make somethings pass. I appreciate there are some things we can't remove, but this is superflous.

Reply:
Have removed this function entirely, replacing it with `if/else`s. It's only used in one place, so this should be fine.